app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java

```diff
@@ -396,50 +406,122 @@ private void onPhotoReturned(File photoFile) {
             mProduct.setImgupload_front(photoFile.getAbsolutePath());
             imgSaveFront.setVisibility(View.VISIBLE);
             mAttacherimgSaveFront = new PhotoViewAttacher(imgSaveFront);
-            Picasso.with(this)
-                    .load(photoFile)
-                    .into(imgSaveFront, new Callback() {
-                        @Override
-                        public void onSuccess() {
-                            mAttacherimgSaveFront.update();
-                        }
-
-                        @Override
-                        public void onError() {
-                        }
-                    });
+            if (!Utils.isConnectedToMobileData(getApplicationContext())) {
+                Picasso.with(this)
+                        .load(photoFile)
+                        .into(imgSaveFront, new Callback() {
+                            @Override
+                            public void onSuccess() {
+                                mAttacherimgSaveFront.update();
+                            }
+
+                            @Override
+                            public void onError() {
+                            }
+                        });
+            } else {
+                try {
+                    String resolution = mSharedPref.getString("imageUpload", "640 X 480");
+                    int x = Integer.parseInt(resolution.substring(0, resolution.indexOf(" ")));
+                    int y = Integer.parseInt(resolution.substring(resolution.lastIndexOf(" ") + 1));
+                    Picasso.with(this)
+                            .load(photoFile)
+                            .resize(x, y)
+                            .into(imgSaveFront, new Callback() {
+                                @Override
+                                public void onSuccess() {
+                                    mAttacherimgSaveFront.update();
+                                }
+
+                                @Override
+                                public void onError() {
+                                }
+                            });
+                } catch (Exception e){
+                    Toast.makeText(getApplicationContext(), "Please set Resolution First", Toast.LENGTH_SHORT).show();
```

Comment:
This needs to be translateable, and if it is not set, it should have a default setting

Reply:
"Please set a resolution first"